src/core/QueryManager.ts

```diff
@@ -994,7 +1009,8 @@ export class QueryManager<TStore> {
           concast.beforeNext(() => {
             if (byVariables.delete(varJson) &&
                 byVariables.size < 1) {
-              inFlightLinkObservables.delete(serverQuery);
+              const index = inFlightLinkObservables.findIndex(observable => observable.serverQuery !== serverQuery);
+              inFlightLinkObservables.splice(index);
```

Comment:
You'll want to pass in a `deleteCount` to `splice`, otherwise you end up removing everything after the start index.

```suggestion
              inFlightLinkObservables.splice(index, 1);
```

Since tests currently pass, you may want to consider adding a test to ensure in flight observables are removed/kept as expected. What do you think?

Reply:
Oh good catch. Yeah I had hoped that existing tests would catch something like that! I'll see what I can do.

Reply:
oh also my findIndex isn't quite right.. I think I am going to implement this with .filter after all. It's simpler overall and I don't think there is really any performance difference.

Reply:
Okay, I adjusted the logic such that it should be right. I am not sure how to set up a test that will make sure we don't have regressions on this cleanup however. I can try and figure it out at a later time. For now I'm feeling done with this.